twisted/internet/asyncioreactor.py

```diff
@@ -171,18 +171,18 @@ def addWriter(self, writer):
                                               self._readOrWrite, writer,
                                               False)
             self._writers[writer] = fd
-        except IOError as e:
+        except PermissionError:
             self._unregisterFDInAsyncio(fd)
-            if e.errno == errno.EPERM:
-                # epoll(7) doesn't support certain file descriptors,
-                # e.g. filesystem files, so for those we just poll
-                # continuously:
-                self._continuousPolling.addWriter(writer)
-            else:
-                raise
-        except OSError as e:
+            # epoll(7) doesn't support certain file descriptors,
+            # e.g. filesystem files, so for those we just poll
+            # continuously:
+            self._continuousPolling.addWriter(writer)
+        except BrokenPipeError:
             # The kqueuereactor will raise this if there is a broken pipe
```

Comment:
I don't like it either, but there's no good way (that I'm aware of) for Python to express "for exceptions A, B, C, do this common logic, but for C only, do this extra stuff" without isinstance checks.  Would you prefer an isinstance check?